server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -0,0 +1,155 @@
+        Method method = request.getHttpRequest().method();
+        String uri = request.getHttpRequest().uri();
+        if (uri.startsWith(uriPrefix)) {
+            uri = uri.substring(uriPrefix.length());
+        }
+        String message = "Forwarding the request " + method + " " + uri + " to " + discoveryExtension;
+        logger.info(message);
+        // Hack to pass a final class in to fetch the response string
+        final StringBuilder responseBuilder = new StringBuilder();
+        final CountDownLatch inProgressLatch = new CountDownLatch(1);
+        final TransportResponseHandler<RestExecuteOnExtensionResponse> restExecuteOnExtensionResponseHandler = new TransportResponseHandler<
+            RestExecuteOnExtensionResponse>() {
+
+            @Override
+            public RestExecuteOnExtensionResponse read(StreamInput in) throws IOException {
+                return new RestExecuteOnExtensionResponse(in);
+            }
+
+            @Override
+            public void handleResponse(RestExecuteOnExtensionResponse response) {
+                responseBuilder.append(response.getResponse());
+                logger.info("Received response from extension: {}", response.getResponse());
+                inProgressLatch.countDown();
+            }
+
+            @Override
+            public void handleException(TransportException exp) {
+                responseBuilder.append("FAILED: ").append(exp);
+                logger.debug(new ParameterizedMessage("REST request failed"), exp);
+                inProgressLatch.countDown();
+            }
+
+            @Override
+            public String executor() {
+                return ThreadPool.Names.GENERIC;
+            }
+        };
+        try {
+            transportService.sendRequest(
+                discoveryExtension,
+                ExtensionsOrchestrator.REQUEST_REST_EXECUTE_ON_EXTENSION_ACTION,
+                new RestExecuteOnExtensionRequest(method, uri),
+                restExecuteOnExtensionResponseHandler
+            );
+            try {
+                inProgressLatch.await(5, TimeUnit.SECONDS);
+            } catch (InterruptedException e) {
+                return channel -> channel.sendResponse(
+                    new BytesRestResponse(RestStatus.REQUEST_TIMEOUT, "No response from extension to request.")
+                );
+            }
+        } catch (Exception e) {
+            logger.info("Failed to send REST Actions to extension " + discoveryExtension.getName(), e);
+        }
+        String response = responseBuilder.toString();
+        if (response.isBlank() || response.startsWith("FAILED")) {
+            return channel -> channel.sendResponse(
+                new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, response.isBlank() ? "Request Failed" : response)
+            );
+        }
+        return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.OK, response));
```

Comment:
Thanks for this @dbwiddis. 
You've suggested an improvement to not let the user wait, could you open an issue if you haven't already to discuss a way to optimize it?

Reply:
I'll think about it.  We probably need to implement more RestActions and see how long-running jobs are already handled, so I'm thinking it'll pop up later when it's needed and we can address it then, along with many other incremental changes we'll need.

Reply:
Really I think this is part of a bigger meta-issue regarding initialization, reachability, etc. [SDK #94](https://github.com/opensearch-project/opensearch-sdk-java/issues/94) touches on some related issues and it can probably be discussed there.